api/src/State/PersonalInvitationAcceptProcessor.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace App\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProcessorInterface;
+use App\DTO\PersonalInvitation;
+use App\Entity\CampCollaboration;
+use App\Entity\User;
+use App\Repository\CampCollaborationRepository;
+use App\Repository\UserRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\NonUniqueResultException;
+use Doctrine\ORM\NoResultException;
+use Symfony\Bundle\SecurityBundle\Security;
+
+/**
+ * @implements ProcessorInterface<PersonalInvitation,PersonalInvitation>
+ */
+class PersonalInvitationAcceptProcessor implements ProcessorInterface {
+    public function __construct(
+        private readonly CampCollaborationRepository $campCollaborationRepository,
+        private readonly UserRepository $userRepository,
+        private readonly Security $security,
+        private readonly EntityManagerInterface $em,
+    ) {}
+
+    /**
+     * @param PersonalInvitation $data
+     *
+     * @throws NoResultException
+     * @throws NonUniqueResultException
+     */
+    public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = []): PersonalInvitation {
+        $user = $this->getUser();
+
+        $campCollaboration = $this->campCollaborationRepository->findByUserAndIdAndInvited($user, $data->id);
+        $campCollaboration->status = CampCollaboration::STATUS_ESTABLISHED;
+        $campCollaboration->inviteKey = null;
+        $campCollaboration->inviteKeyHash = null;
+        $campCollaboration->inviteEmail = null;
+
+        $this->em->flush();
+
+        return $data;
+    }
+
+    /**
+     * @throws NonUniqueResultException
+     * @throws NoResultException
+     */
+    protected function getUser(): ?User {
+        $user = $this->security->getUser();
+        if (null == $user) {
+            return null;
```

Comment:
If you don't handle the null case, it would be wiser to throw a validationException that you are not authenticated.
That way you don't have NPE later

Reply:
[2fbe07b](https://github.com/ecamp/ecamp3/pull/4865/commits/2fbe07b4ae9a907eed17f60b7c009e026308e89f)
But this can never happen, because of our security rules on the API operation. So I was not able to write a test for covering this.